.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,23 @@
+name: Run build/tests/lint on pull requests
+
+on:
+  pull_request:
+
+# By default the karma test runners use the karma 'Chrome' runner
+# This is great when running locally because the browser pops up and you get to see what it does
+# This doesn't work in Actions, however, so we set this env var to force the Headless runner
+env:
+  TEST_BROWSERS: ChromeHeadless
```

Comment:
Are we sure all tests work with headless browser?

Reply:
Looks like they do: You can verify in the checks on this PR.